sdks/RepoToolset/tools/DefaultVersions.props

```diff
@@ -49,10 +55,18 @@
   <Import Project="$(DotNetPackageVersionPropsPath)" Condition="'$(DotNetPackageVersionPropsPath)' != ''" />
   <Import Project="$(FixedVersionsPropsPath)" Condition="'$(FixedVersionsPropsPath)' != ''"/>
 
-  <PropertyGroup>
+  <!-- RestoreSources overrides - defines DotNetRestoreSources variable if available -->
+  <Import Project="$(DotNetRestoreSourcePropsPath)" Condition="'$(DotNetRestoreSourcePropsPath)' != ''"/>
+
+  <!-- Force sources to DotNetRestoreSources if specified, ignore any sources set by the repo -->
+  <PropertyGroup Condition="'$(DotNetRestoreSources)' != ''">
+    <RestoreSources>$(DotNetRestoreSources)</RestoreSources>
```

Comment:
If there are sources set by the repo, DotNetRestoreSources shouldn't always override them.  If we're building online, DotNetRestoreSources should be the first, but others need to be present to allow repos to find their packages in the online build.  For offline build, DotNetRestoreSources should be the only one.

Reply:
OK, I'll add the offline check.